### PR TITLE
[oxygen] Merge forward from 2017.7 to oxygen

### PR DESCRIPTION
**Pull request by rallytime**
Conflicts:
  - salt/modules/gnomedesktop.py
  - salt/utils/cloud.py

### CODE REVIEW

File: salt/modules/gnomedesktop.py [hunk 1]
```diff
@@ -50,6 +52,17 @@ def __init__(self, user, schema, key):
         self.UID = None
         self.HOME = None
 
+    @property
+    def gestting_command(self):
```
Comment (terminalmage) [inline salt/modules/gnomedesktop.py]: I'm guessing this was supposed to be `gsetting_command`, not `gestting_command`.

File: salt/modules/gnomedesktop.py [hunk 2]
```diff
@@ -62,7 +75,7 @@ def _get(self):
             log.info('User does not exist')
             return False
 
-        cmd = 'dbus-launch --exit-with-session gsettings get {0} {1}'.format(self.SCHEMA, self.KEY)
+        cmd = self.gestting_command + ['get', str(self.SCHEMA), str(self.KEY)]
```
Comment (terminalmage) [inline salt/modules/gnomedesktop.py]: Same misspelling here.

File: salt/modules/gnomedesktop.py [hunk 3]
```diff
@@ -90,8 +103,7 @@ def _set(self, value):
             result['stdout'] = 'User {0} does not exist'.format(user)
             return result
 
-        cmd = 'dbus-launch --exit-with-session gsettings set {0} {1} "{2}"'.format(
-                self.SCHEMA, self.KEY, value)
+        cmd = self.gestting_command + ['set', self.SCHEMA, self.KEY, value]
```
Comment (terminalmage) [inline salt/modules/gnomedesktop.py]: Also here.

**Review by terminalmage — CHANGES_REQUESTED** [review]
Some misspellings.

**Review by terminalmage — APPROVED** [review]
Actually nevermind, this is a merge, I will fix the things I addressed in my review in the proper branch.

**Comment by terminalmage** [comment]
Misspellings addressed in https://github.com/saltstack/salt/pull/45773

**Comment by rallytime** [comment]
@terminalmage Good catch! I merged your 2017.7 changes in and merged those forward to this PR.

**Pull request merged**